### DISCUSSION TITLE
defineChain overwrites fields

### DISCUSSION THREAD

**Discussion by aaronmgdr** [wevm/viem#816] [Idea / Feature Request]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Package Version

1.1.7

### Current Behavior

if i call the method with a chain to add serializer, than any formatters on the chain are set to undefined

```
defineChain(optimisim, {serializers: {transaction}})

```

### Expected Behavior

if the config does just have a field i would expect the output to have whatever was originally on the chain.

**Comment by jxom** [comment]
What do you mean?

**Comment by aaronmgdr** [comment]
```
import { celoAlfajores } from 'viem/chains'
import {defineChain} from 'viem'
import {serializeTransaction} from '../serializeTransaction'

//IF i dont explicitly add formatters to the config here they are removed from the chain. 
const celoWithSerializer = defineChain(celoAlfajores, {formatters: celoAlfajores.formatters, serializers: {transaction: serializeTransaction}})

celoAlfajores.formatters === celoWithSerializer.formatters /// true
```

This was unexpected for me. I would have thought this and the above line would be the same 
```
const celoWithSerializer = defineChain(celoAlfajores, {serializers: {transaction: serializeTransaction}})

celoWithSerializer.formatters === celoAlfajores.formatters // currently this is false
```

**Comment by aaronmgdr** [comment]
if i'm off here in my expectations.  let me know.

**Comment by jxom** [comment]
Yeah, type inference starts to get a bit weird when you try to start composing it like that. :\

Let's move this to a discussion, and we can improve on it in the future.
